src/workflows/steps.js

```diff
@@ -468,3 +470,152 @@ export function setPromote( state ) {
 }
+
+export function getPackageScope( state ) {
+	if ( state.qa && typeof state.qa === "boolean" ) {
+		state.qa = "@lk";
+	} else if ( state.qa && state.qa.charAt( 0 ) !== "@" ) {
+		state.qa = `@${ state.qa }`;
+	}
+
+	return Promise.resolve();
+}
+
+export function getScopedRepos( state ) {
+	const { configPath, qa: scope } = state;
+	let content = {};
+	try {
+		content = util.readJSONFile( configPath );
+
+		let repos = Object.keys( content.devDependencies ).filter( key => key.includes( scope ) );
+		repos = repos.concat( Object.keys( content.dependencies ).filter( key => key.includes( scope ) ) );
+		repos = repos.map( key => key.slice( ( scope.length + 1 ), key.length ) );
+
+		if ( repos.length === 0 ) {
+			util.advise( "noPackagesInScope" );
+			process.exit( 0 ); // eslint-disable-line no-process-exit
+		}
+
+		return Promise.resolve( repos );
+	} catch ( err ) {
+		util.advise( "updateVersion" );
+	}
+
+	return Promise.resolve();
+}
+
+export function askReposToUpdate( state ) {
+	return getScopedRepos( state ).then( packages => {
+		return util.prompt( [ {
+			type: "checkbox",
+			name: "packagesToPromote",
+			message: "Select the package(s) you wish to update:",
+			choices: packages
+		} ] ).then( ( { packagesToPromote } ) => {
+			state.packages = packagesToPromote;
+			return Promise.resolve();
+		} );
+	} );
+}
+
+export function askVersion( pkg ) {
+	return () => util.prompt( [ {
+		type: "input",
+		name: "version",
+		message: `What version do you want to update ${ pkg } to:`
+	} ] ).then( ( { version } ) => {
+		return Promise.resolve( { pkg, version } );
+	} );
+}
+
+export function askVersions( state ) {
+	const { packages } = state;
+	const prompts = packages.map( pkg => askVersion( pkg ) );
+
+	return sequence( prompts ).then( dependencies => {
+		state.dependencies = dependencies;
+		return Promise.resolve();
+	} );
+}
+
+export function askChangeType( state ) {
+	return util.prompt( [ {
+		type: "list",
+		name: "changeType",
+		message: "What type of change is this work",
+		choices: [ "feature", "defect", "rework" ]
+	} ] ).then( ( { changeType } ) => {
+		state.changeType = changeType;
+		return Promise.resolve();
+	} );
+}
+
+export function askChangeReason( state ) {
+	return util.prompt( [ {
+		type: "input",
+		name: "changeReason",
+		message: "What is the reason for this change"
+	} ] ).then( ( { changeReason } ) => {
+		state.changeReason = changeReason;
+		return Promise.resolve();
+	} );
+}
+
+export function gitCheckoutBranch( state ) {
+	const branch = `${ state.changeType }-${ removeWords( state.changeReason ).join( "-" ) }`;
+	return git.checkoutBranch( branch );
+}
+
+export function updateDependencies( state ) {
+	const { dependencies, configPath, qa: scope } = state;
+	let content = {};
+	try {
+		content = util.readJSONFile( configPath );
+
+		dependencies.forEach( item => {
+			const key = `${ scope }/${ item.pkg }`;
+			if ( key in content.devDependencies ) {
```

Comment:
same concern here that `devDependencies` or `dependencies` could be `null` or `undefined` (but honestly probably not for our projects)